lib/configuration.dart

```diff
@@ -0,0 +1,93 @@
+import 'constants.dart';
+import 'tracking_options.dart';
+import 'default_tracking.dart';
+
+/// Configuration for Amplitude instance.
+/// 
+/// Before initialzing Amplitude instance, create a Configuration instance
+/// with your desired configuration and pass it to the Amplitude instance.
+/// Notice that only `optOut` and be changed later by calling `setOptOut()`.
+/// Other configurations can *NOT* be changed after initialzing Amplitude instance.
+class Configuration {
+  String apiKey;
+  int flushQueueSize;
+  int flushIntervalMillis;
+  late String instanceName;
+  bool optOut;
+  LogLevel logLevel;
+  int? minIdLength;
+  String? partnerId;
+  int flushMaxRetries;
+  bool useBatch;
+  ServerZone serverZone;
+  String? serverUrl;
+  int minTimeBetweenSessionsMillis;
+  DefaultTrackingOptions defaultTracking;
+  TrackingOptions trackingOptions;
+  // Mobile (iOS and Android) specific
+  bool enableCoppaControl;
+  bool flushEventsOnClose;
+  int identifyBatchIntervalMillis;
+  bool migrateLegacyData;
+  // Android specific
+  bool locationListening;
+  bool useAdvertisingIdForDeviceId;
+  bool useAppSetIdForDeviceId;
```

Comment:
For clarity maybe we should move these to an Adnroid specific part of the config to make it clear they are not supported in other platforms.

e.g.
```
class AndroidConfiguration {
  bool useAppSetIdForDeviceId;
}

class Configuration {
  ...
  AndroidConfiguration android;
}
```

Reply:
Genius! I love it! Updated in https://github.com/amplitude/Amplitude-Flutter/pull/162/commits/455c90a052574939519a0ecbd2428dbfd57ab734

Example
```
var configuration = Configuration(
  apiKey: "apiKey",
  mobileConfiguration: MobileConfiguration(
    enableCoppaControl: true,
    flushEventsOnClose: false
  ),
  androidConfiguration: AndroidConfiguration(
    locationListening: false,
    useAdvertisingIdForDeviceId: true,
    useAppSetIdForDeviceId: true
  ),
);
```

Reply:
Hey @Mercy811, sorry to change my mind on this but after seeing this broken up by platform, I am thinking the original "merged configuration" is probably better.

1. Some options that are only supported in Android, etc may be added to the other SDKs in the future. e.g. locationListening could maybe be added to iOS.
 
To make it so we can add platform support without modifying the config structure, it is probably be to keep all the options at the same level in the config, and clearly define platform support in code comments. Then we can update the comments as we add support to other platforms.

e.g.
```
  // Supported on Android only.
  useAdvertisingIdForDeviceId: true,
```